### PR TITLE
AP_DDS: publish gps global origin

### PR DESCRIPTION
**Pull request by srmainwaring**
Add message type `geographic_msgs/msg/GeoPointStamped` and publish the GPS global origin at 1 Hz. 

Uses the same logic as the mavlink function `GCS_MAVLINK::send_gps_global_origin()`. 

### Details

- Datatype: `geographic_msgs/msg/GeoPointStamped`
- Topic: `/ap/gps_global_origin/filtered`

Relevant section in [REP 147](https://ros.org/reps/rep-0147.html): [GPS Messages](https://ros.org/reps/rep-0147.html#gps-messsages)

### Testing

Launch SITL with DDS for quad copter:

```bash
% ros2 launch ardupilot_sitl sitl_dds_udp.launch.py transport:=udp4 refs:=$(ros2 pkg prefix ardupilot_sitl)/share/ardupilot_sitl/config/dds_xrce_profile.xml synthetic_clock:=True wipe:=False model:=quad speedup:=1 slave:=0 instance:=0 defaults:=$(ros2 pkg prefix ardupilot_sitl)/share/ardupilot_sitl/config/default_params/copter.parm,$(ros2 pkg prefix ardupilot_sitl)/share/ardupilot_sitl/config/default_params/dds_udp.parm sim_address:=127.0.0.1 master:=tcp:127.0.0.1:5760 sitl:=127.0.0.1:5501
```

New topic visible in the node:

```bash
$ ros2 node info /ardupilot_dds
/ardupilot_dds
  Subscribers:
    /ap/cmd_gps_pose: ardupilot_msgs/msg/GlobalPosition
    /ap/cmd_vel: geometry_msgs/msg/TwistStamped
    /ap/joy: sensor_msgs/msg/Joy
    /ap/tf: tf2_msgs/msg/TFMessage
  Publishers:
    /ap/battery/battery0: sensor_msgs/msg/BatteryState
    /ap/clock: rosgraph_msgs/msg/Clock
    /ap/geopose/filtered: geographic_msgs/msg/GeoPoseStamped
    /ap/gps_global_origin/filtered: geographic_msgs/msg/GeoPointStamped
    /ap/imu/experimental/data: sensor_msgs/msg/Imu
    /ap/navsat/navsat0: sensor_msgs/msg/NavSatFix
    /ap/pose/filtered: geometry_msgs/msg/PoseStamped
    /ap/tf_static: tf2_msgs/msg/TFMessage
    /ap/time: builtin_interfaces/msg/Time
    /ap/twist/filtered: geometry_msgs/msg/TwistStamped
  Service Servers:
    /ap/arm_motors: ardupilot_msgs/srv/ArmMotors
    /ap/mode_switch: ardupilot_msgs/srv/ModeSwitch
  Service Clients:

  Action Servers:

  Action Clients:
```

Echo topic:

```bash
$ ros2 topic echo /ap/gps_global_origin/filtered --once
header:
  stamp:
    sec: 1709667425
    nanosec: 87921000
  frame_id: base_link
position:
  latitude: -35.36326217651367
  longitude: 149.1652374267578
  altitude: 584.0899658203125
---
```

### Flash

No change.

#### Before (2d5e6a5f98)

```bash
Target               Text (B)  Data (B)  BSS (B)  Total Flash Used (B)  Free Flash (B)  External Flash Used (B)
---------------------------------------------------------------------------------------------------------------
bin/antennatracker    3358720    245760        0               3604480  Not Applicable  Not Applicable         
bin/arducopter        4390912    278528        0               4669440  Not Applicable  Not Applicable         
bin/arducopter-heli   4407296    278528        0               4685824  Not Applicable  Not Applicable         
bin/arduplane         4325376    278528        0               4603904  Not Applicable  Not Applicable         
bin/ardusub           3899392    262144        0               4161536  Not Applicable  Not Applicable         
bin/blimp             3424256    245760        0               3670016  Not Applicable  Not Applicable         
bin/ardurover         4063232    278528        0               4341760  Not Applicable  Not Applicable
```

#### After (1ffb2c3175)

```bash
Target               Text (B)  Data (B)  BSS (B)  Total Flash Used (B)  Free Flash (B)  External Flash Used (B)
---------------------------------------------------------------------------------------------------------------
bin/antennatracker    3358720    245760        0               3604480  Not Applicable  Not Applicable         
bin/arducopter        4390912    278528        0               4669440  Not Applicable  Not Applicable         
bin/arducopter-heli   4407296    278528        0               4685824  Not Applicable  Not Applicable         
bin/arduplane         4325376    278528        0               4603904  Not Applicable  Not Applicable         
bin/ardusub           3899392    262144        0               4161536  Not Applicable  Not Applicable         
bin/blimp             3424256    245760        0               3670016  Not Applicable  Not Applicable         
bin/ardurover         4063232    278528        0               4341760  Not Applicable  Not Applicable         
```

### CODE REVIEW

File: libraries/AP_DDS/dds_xrce_profile.xml [hunk 1]
```diff
@@ -246,6 +246,34 @@
...
+    <historyQos>
+      <kind>KEEP_LAST</kind>
+      <depth>5</depth>
+    </historyQos>
+  </topic>
+  <data_writer profile_name="gps_global_origin__dw">
+    <historyMemoryPolicy>PREALLOCATED_WITH_REALLOC</historyMemoryPolicy>
+    <qos>
+      <reliability>
+        <kind>BEST_EFFORT</kind>
```
[the remark this answers is not in the document]
Comment (srmainwaring) [reply libraries/AP_DDS/dds_xrce_profile.xml:261]: Should be set to best effort stream, same as `/ap/geopose/filtered`. We don't use reliable streams for many topics.

[the remark this answers is not in the document]
Comment (Ryanf55) [reply libraries/AP_DDS/dds_xrce_profile.xml:261]: Thanks for the details. I can test tonight, overall LGTM!

**Review by Ryanf55 — APPROVED** [review]
LGTM. 

1) Can you update the README with the new topic
2) If you have time, can you add a ROS test to make sure it's being published?

**Comment by srmainwaring** [comment]
> 1. Can you update the README with the new topic

Updated in https://github.com/ArduPilot/ardupilot/pull/26409/commits/3d0d9a7c5188fd4650617b87002896b5885f2f5d

**Review by Ryanf55 — APPROVED** [review]
Approved as it meets the minimum requirements for submission. If you add a test, CI can check the code is correct.

**Comment by srmainwaring** [comment]
> Approved as it meets the minimum requirements for submission. If you add a test, CI can check the code is correct.

Thanks @Ryanf55 - can you please label as merge in CI pass.

**Pull request merged**